src/ui/win32/Desktop.cpp

```diff
@@ -0,0 +1,63 @@
+#include "ui/win32/Desktop.hh"
+
+// Win32 specific implementation of Desktop.hh
+
+#include "ui/win32/Dlg_MessageBox.hh"
+
+#include <assert.h>
+
+namespace ra {
+namespace ui {
+namespace win32 {
+
+Desktop::Desktop() noexcept
+{
+    m_vDialogControllers.push_back(new Dlg_MessageBox::Controller());
```

Comment:
A controller maps a ViewModel to an implementation-specific window. When `Show` or `ShowWindow` is called, the appropriate controller is called to generate the UI.

Reply:
I thought this was supposed to be MVVM and not MVC? Not that I'm complaining though, just a bit confused.

Reply:
It sounds like a router. Not sure though.

Reply:
I've renamed these to `Presenter`

Reply:
Seems like a better name.